src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -715,11 +803,26 @@ private TcpConnectionInfo GetConnectionInfo(ReplicationNode node, bool external)
             {
                 if (node is ExternalReplication exNode)
                 {
-                    using (var requestExecutor = RequestExecutor.Create(exNode.ConnectionString.TopologyDiscoveryUrls, exNode.ConnectionString.Database, _server.Server.Certificate.Certificate, DocumentConventions.Default))
+                    var database = exNode.ConnectionString.Database;
+                    var certificate = GetCertificateForReplication(exNode, out _);
+
+                    if (exNode.PullReplicationAsEdgeSettings == null)
+                    {
+                        using (var requestExecutor = RequestExecutor.Create(exNode.ConnectionString.TopologyDiscoveryUrls, exNode.ConnectionString.Database, certificate, DocumentConventions.Default))
+                        using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                        {
+                            var cmd = new GetTcpInfoCommand("external-replication", database);
+                            requestExecutor.Execute(cmd, ctx);
+                            node.Database = database;
+                            node.Url = requestExecutor.Url;
+                            return new []{ cmd.Result };
+                        }
+                    }
+                   
+                    using (var requestExecutor = RequestExecutor.CreateForFixedTopology(exNode.ConnectionString.TopologyDiscoveryUrls, exNode.ConnectionString.Database, certificate, DocumentConventions.Default))
```

Comment:
Can save some code by unifying the `requestExecutor ` and `ctx` usages across the `if`s